bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationService.java

```diff
@@ -201,10 +201,11 @@ public void handleEvent(@Nullable Event event) {
             initParams.put("com.sun.jersey.api.json.POJOMappingFeature", "false");
             initParams.put(ServletProperties.PROVIDER_WEB_APP, "false");
             httpService.registerServlet(RESTAPI_PATH, new ServletContainer(resourceConfig), initParams, null);
-            if (discovery == null) {
+            UpnpServer localDiscovery = discovery;
```

Comment:
Just a question: Why don't you use
```
            UpnpServer discovery = this.discovery;
```

IMO this is more clear.

Reply:
I'm used to SonarQube telling me declaring local variables that hide fields is a bad practice. ;-)

Reply:
Ok 👍

Reply:
This is the rule I'm using on other projects. It sometimes saves you from creating bugs too.

http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.html